### ISSUE TITLE
Debian package?

### ISSUE THREAD

**Issue by Kreeblah** [topgrade-rs/topgrade#897]
## I want to suggest some general feature
It would be nice if there were a Debian package available for this (even if it requires adding another repo).  Currently, Debian stable doesn't have a new enough version of rust in its packaging system to build the project (and, given their focus is on stability rather than the newest versions of things, I wouldn't expect that to change), so installing it requires installing binaries the package manager doesn't know about.  Even installing `cargo-binstall` doesn't work, as it requires a newer version of `rustc` than Debian offers.

It'd be much easier if there were a package that could just install the binary on its own, rather than having to build it on a host and then copy it over to other hosts every time there's an update.

**Comment by SteveLauC** [comment]
> Currently, Debian stable doesn't have a new enough version of rust in its packaging system to build the project

What errors did you get when you tried to build Topgrade with that version of Rust toolchain?🤔

> It would be nice if there were a Debian package available for this

Packaging for Linux is hard, so I won't do this. But I am willing to help if someone wants to get this done.

**Comment by Kreeblah** [comment]
> > Currently, Debian stable doesn't have a new enough version of rust in its packaging system to build the project
> 
> What errors did you get when you tried to build Topgrade with that version of Rust toolchain?🤔

It fails on one of the dependencies, because the provided version of `rustc` is 1.63.0.

```
error: failed to compile `topgrade v15.0.0`, intermediate artifacts can be found at `/tmp/cargo-install7Fo7de`

Caused by:
  package `roff v0.2.2` cannot be built because it requires rustc 1.70 or newer, while the currently active rustc version is 1.63.0
  Try re-running cargo install with `--locked`
```

> > It would be nice if there were a Debian package available for this
> 
> Packaging for Linux is hard, so I won't do this. But I am willing to help if someone wants to get this done.

I might take a look at this when I have time, then.  I've never built a Debian package before, but I've built packages for other OSes (like FreeBSD), so I imagine I'll be able to figure it out somehow and get something set up as a GitHub action.

**Comment by SteveLauC** [comment]
> It fails on one of the dependencies, because the provided version of rustc is 1.63.0.

Looks like Topgrade's MSRV is 1.70.

> I might take a look at this when I have time,

Appreciate it, feel free to ask me for help if needed!

**Comment by SteveLauC** [comment]
#898 adds the `.deb` files to the release assets, though we are currently not adding them to the official Debian repos, 2 issues remain to be solved:

1. Topgrade needs Rust toolchain 1.70 or higher, the Rust toolchain packaged by Debian is 1.63.0, we don't know if Debian will accept tools that can't be built with available Debian packages.
2. Debian packages need a changelog, Topgrade does not have changelogs.

For more information, see the discussion [there](https://github.com/topgrade-rs/topgrade/pull/898#issuecomment-2327811682).

**Comment by SteveLauC** [comment]
> Looks like Topgrade's MSRV is 1.70.

This is incorrect, Topgrade needs Rust 1.76.0 or higher, I just pinned the toolchain to the MSRV in #900.

**Comment by Kreeblah** [comment]
Ah, nice.  That'll help identify when it need to be updated in the documentation.

**Comment by gmkey** [comment]
Just checking: are you aware that the release contains Debian packages? 
[topgrade_16.0.2-1_amd64.deb](https://github.com/topgrade-rs/topgrade/releases/download/v16.0.2/topgrade_16.0.2-1_amd64.deb)
[topgrade_16.0.2-1_arm64.deb](https://github.com/topgrade-rs/topgrade/releases/download/v16.0.2/topgrade_16.0.2-1_arm64.deb)

and I have created a build request to build an armv7 Debian package.

**Comment by SteveLauC** [comment]
> are you aware that the release contains Debian packages?

The related CI scripts were added by @Kreeblah in #898

**Comment by GideonBear** [comment]
FYI: [deb-get](https://github.com/wimpysworld/deb-get) has support for Topgrade now since https://github.com/wimpysworld/deb-get/pull/1398. It technically doesn't meet your requirement of "even if it requires adding another repo", but it should solve your problem if you're okay with installing `deb-get`.

**Comment by SteveLauC** [comment]
@GideonBear Thanks for reaching out to them!
